Don't try to compile gcocoanotificationbackend.c on OS X < 10.9 which
doesn't support it.
https://bugzilla.gnome.org/show_bug.cgi?id=794380
--- gio/Makefile.am.orig	2018-03-12 11:23:37.000000000 -0500
+++ gio/Makefile.am	2018-03-15 14:27:31.000000000 -0500
@@ -612,8 +612,11 @@
 	gnextstepsettingsbackend.c \
 	gosxcontenttype.c \
 	gosxappinfo.c \
-	gosxappinfo.h \
+	gosxappinfo.h
+if MAC_OS_X_9
+libgio_objc_2_0_la_SOURCES += \
 	gcocoanotificationbackend.c
+endif
 
 noinst_LTLIBRARIES += libgio-objc-2.0.la
 libgio_2_0_la_LIBADD += libgio-objc-2.0.la
